scene/gui/line_edit.cpp

```diff
@@ -1777,7 +1777,7 @@ Size2 LineEdit::get_minimum_size() const {
 
 	if (expand_to_text_length) {
 		// Add a space because some fonts are too exact, and because caret needs a bit more when at the end.
-		min_size.width = MAX(min_size.width, full_width + em_space_size);
```

Comment:
Comment and the code is ancient (https://github.com/godotengine/godot/commit/405f6af79c32c575768f53661127cfd6aa4230f4), and not necessarily true (it's possible it was added to fix rounding errors of the old Font implementation). A width of full space should not be necessary to fit the caret, but it still should be tested with multiple fonts.

Reply:
Without it, me, as Godot user, can't remove the empty space and make my LineEdit fully fits text.
In case of comment: if lack of the empty space provides some buggy behavior, you can just add this space with theme properties or LineEdit's minimum_character_width property